src/components/Card.js

```diff
@@ -1,37 +1,63 @@
 export default class Card {
-  constructor({ location, link }, cardSelector, handleImageClick) {
-    this._location = location;
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteConfirmSubmit,
+    handleCardLike
+  ) {
+    this._name = name;
     this._link = link;
+    this.cardId = _id;
+    this.isLiked = isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteConfirmSubmit = handleDeleteConfirmSubmit;
+    this._handleCardLike = handleCardLike;
+    this._confirmDeleteImagePopup = document.querySelector(
+      "#confirm-image-delete-popup"
+    );
     this._data = {
-      location: this._location,
+      location: this._name,
       link: this._link,
     };
   }
 
   _setEventListeners() {
     // card__like-button
     this._cardLikeButton.addEventListener("click", () => {
-      this._handleLikeIcon();
+      this._handleCardLike(this);
     });
     // card__delete-button
     this._cardDeleteIcon.addEventListener("click", () => {
-      this._handleDeleteCard();
+      this._handleDeleteConfirmSubmit(this);
     });
 
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick(this._data);
     });
   }
 
-  _handleLikeIcon() {
-    this._cardLikeButton.classList.toggle("card__like-button_active");
-  }
-
-  _handleDeleteCard() {
+  handleDeleteCard() {
```

Comment:
Needs correction: please, rename this method to `deleteCard` since it's not an event handler anymore